### PR TITLE
New OAuthApi Client

### PR DESCRIPTION
**Pull request by alexjoybc**
# Description

This PR includes a new  OAuthApiClient to connect to OAuth provider and get refresh token.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tested locally and got token back from provider

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,16 @@
     └── openshift               # Build and Deploy Configurations.
     └── bin                     # Useful scripts.
 
+## RUN
+
+Create a `.env` file from the `.env.template`
+
+Confiture the necessary variables in `.env`
```
Comment (peggy-gov-bc-ca) [inline README.md]: type "configure'

File: app/DynamicsAdapter/DynamicsAdapter.Web/Auth/OAuthApiClient.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            var content = new FormUrlEncodedContent(data);
+
+            using (var request = new HttpRequestMessage(HttpMethod.Post, _oAuthOptions.OAuthUrl) {Content = content})
+            {
+                var response = await _httpClient.SendAsync(request, HttpCompletionOption.ResponseHeadersRead,
+                    cancellationToken);
+
+                if (!response.IsSuccessStatusCode)
+                {
+                    var responseData = response.Content == null
```
Comment (peggy-gov-bc-ca) [inline app/DynamicsAdapter/DynamicsAdapter.Web/Auth/OAuthApiClient.cs]: interesting

Comment (alexjoybc) [reply app/DynamicsAdapter/DynamicsAdapter.Web/Auth/OAuthApiClient.cs]: that's default from dotnet httpclient

File: README.md (hunk 1 above)
Comment (alexjoybc) [reply README.md]: lol confiture means jam in french

Comment (peggy-gov-bc-ca) [reply README.md]: good to know. It is up to you to change it or not.

**Pull request merged**